sympy/tensor/array/dense_ndim_array.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import print_function, division
+import functools
+from sympy import sympify, Matrix, flatten
+from sympy.tensor.array.mutable_ndim_array import MutableNDimArray
+from sympy.tensor.array.ndim_array import NDimArray
+
+
+class DenseNDimArray(NDimArray):
+
+    def __getitem__(self, index):
+        """
+        Allows to get items from N-dim array.
+
+        Examples
+        ========
+
+        >>> from sympy.tensor.array import MutableDenseNDimArray
+        >>> a = MutableDenseNDimArray(2, 2, [0, 1, 2, 3])
+        >>> a
+        [[0, 1], [2, 3]]
+        >>> a[0, 0]
+        0
+        >>> a[1, 1]
+        3
+
+        """
+        index = self._parse_index(index)
+        return self._array[index]
+
+    @classmethod
+    def zeros(cls, *shape):
+        list_length = functools.reduce(lambda x, y: x*y, shape)
+        return cls._new(*(shape + ([0]*list_length,)))
+
+    def tomatrix(self):
+        """
+        Converts MutableDenseNDimArray to Matrix. Can convert only 2-dim array, else will raise error.
+
+        Examples
+        ========
+
+        >>> from sympy.tensor.array import MutableDenseNDimArray
+        >>> a = MutableDenseNDimArray(3, 3, [1 for i in range(9)])
+        >>> b = a.tomatrix()
+        >>> b
+        Matrix([
+        [1, 1, 1],
+        [1, 1, 1],
+        [1, 1, 1]])
+
+        """
+        if self.rank() != 2:
+            raise ValueError('Dimensions must be of size of 2')
```

Comment:
Perhaps this could say more explicitly that a matrix is expected to be two-dimensional.